tailor_image/create_image.py

```diff
@@ -210,18 +206,12 @@ def update_image_index(release_track, apt_repo, common_config, image_name):
         # Wait for file to be ready to write
         wait_for_index(s3, apt_repo, index_key)
         data = json.load_s3(index_key)
-        if release_label not in data:
-            data[release_label] = base_data[release_label]
-        elif flavour not in data[release_label]:
-            data[release_label][flavour] = base_data[release_label][flavour]
-        else:
-            # If release_label and flavour already exists, update the image and add checksum
-            data[release_label][flavour]['distributions'][distribution] = image_name
-            data[release_label][flavour]['checksums'][image_name] = checksum
     except botocore.exceptions.ClientError as error:
         # If file doesn't exists, we'll create a new one
         if error.response['Error']['Code'] == 'NoSuchKey':
-            data = base_data
+            click.echo('Index file doesn\'t exist, creating a new one')
+
+    data[date_time] = image_data
 
     # Write data to index file
     json.dump_s3(data, index_key)
```

Comment:
I think as I pointed out before, this creates a race conditions that blows out the index file if two images complete building simultaneously. I'd suggest figuring out a way around this, or making the final stage of tailor-image handle that together like tailor-distro does

Reply:
We do have a locking mechanism with the `wait_for_index` method a couple lines before. But I agree that making the final stage handle that would be a better solution.

Reply:
Whoops, missed that one. Onward!